.github/workflows/run_lib_tests.yml

```diff
@@ -0,0 +1,43 @@
+# This workflow will run the appropriate library tests across a python matrix of versions.
+name: Run Python library tests
+
+on:
+  workflow_call
+
+jobs:
+  build_and_test_lib:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ['3.10', '3.11']
+
+    steps:
+      - name: Checkout harmony-browse-image-generator repository
+        uses: actions/checkout@v4
+        with:
+          lfs: true
+
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Install GDAL
+        run: |
+          # Install packaged version of GDAL.
+          sudo apt-get update
+          sudo apt-get install -y libgdal-dev
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install pytest
+          pip install -r pip_requirements.txt
+          # Use the gdal version that was installed
+          pip install GDAL==$(gdal-config --version)
```

Comment:
This was the concession on how I was able to run the tests outside of our docker containers. This is not the same version of gdal that is specified in our docker containers, but in the end, we're only using osgeo's ColorPalette functionality from gdal.

Reply:
That's a good idea because this was actually hard to accomplish.